core/src/main/java/feast/core/grpc/CoreServiceResponseGenerator.java

```diff
@@ -0,0 +1,155 @@
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package feast.core.grpc;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+import feast.core.config.FeastProperties;
+import feast.core.model.Project;
+import feast.core.service.AccessManagementService;
+import feast.core.service.JobService;
+import feast.core.service.SpecService;
+import feast.core.service.StatsService;
+import feast.proto.core.CoreServiceProto.ApplyFeatureSetRequest;
+import feast.proto.core.CoreServiceProto.ApplyFeatureSetResponse;
+import feast.proto.core.CoreServiceProto.ArchiveProjectRequest;
+import feast.proto.core.CoreServiceProto.ArchiveProjectResponse;
+import feast.proto.core.CoreServiceProto.CreateProjectRequest;
+import feast.proto.core.CoreServiceProto.CreateProjectResponse;
+import feast.proto.core.CoreServiceProto.GetFeastCoreVersionResponse;
+import feast.proto.core.CoreServiceProto.GetFeatureSetRequest;
+import feast.proto.core.CoreServiceProto.GetFeatureSetResponse;
+import feast.proto.core.CoreServiceProto.GetFeatureStatisticsRequest;
+import feast.proto.core.CoreServiceProto.GetFeatureStatisticsResponse;
+import feast.proto.core.CoreServiceProto.ListFeatureSetsRequest;
+import feast.proto.core.CoreServiceProto.ListFeatureSetsResponse;
+import feast.proto.core.CoreServiceProto.ListFeaturesRequest;
+import feast.proto.core.CoreServiceProto.ListFeaturesResponse;
+import feast.proto.core.CoreServiceProto.ListIngestionJobsRequest;
+import feast.proto.core.CoreServiceProto.ListIngestionJobsResponse;
+import feast.proto.core.CoreServiceProto.ListProjectsResponse;
+import feast.proto.core.CoreServiceProto.ListStoresRequest;
+import feast.proto.core.CoreServiceProto.ListStoresResponse;
+import feast.proto.core.CoreServiceProto.RestartIngestionJobRequest;
+import feast.proto.core.CoreServiceProto.RestartIngestionJobResponse;
+import feast.proto.core.CoreServiceProto.StopIngestionJobRequest;
+import feast.proto.core.CoreServiceProto.StopIngestionJobResponse;
+import feast.proto.core.CoreServiceProto.UpdateStoreRequest;
+import feast.proto.core.CoreServiceProto.UpdateStoreResponse;
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * Logic for feast core GRPC service. Contains methods that take in Request messages and return
+ * Response messages. This class does not take responsibility for accessManagement.
+ */
+@Component
+public class CoreServiceResponseGenerator {
```

Comment:
I dont understand the purpose of this class. It seems like we need a gRPC service and a REST service, and they can both share services like `SpecService`. Why do we need another layer?

Reply:
I was hoping to abstract out the specific interactions with the different services, in pursuit of consistency between gRPC and REST services (by maintaining a single source of truth for their behaviours). This layer is pretty thin, I created it because I was hesitant to replicate the gRPC implementations into REST, and when one changes the other has to be manually changed as well.

Reply:
What can this class do that `specService` can't do? I mean what functionality would go here that can't go into the `service` classes like `specService`? Seems like we already have a clear abstraction. gRPC/Rest for I/O, services for shared logic.

Reply:
It was an oversight from me due to unfamiliarity with the codebase. Removed the class.